charts/cronjob/templates/cronjob.yaml

```diff
@@ -23,10 +25,8 @@ spec:
           labels:
             app: {{ template "cronjob.name" . }}
             release: {{ .Release.Name }}
-            {{- if eq .Values.isOnDemand }}
+            {{- if .Values.runOnDemandOnly }}
```

Comment:
Should not option and label have the same name, wdyt?

Reply:
As I see it, the label is for filtering so "ondemand:true" sounds better for me than its directive version "runOnDemandOnly". But I am open to any other suggestions as I don't have any strong preference here.

Reply:
It is up to you but i personally find consistent naming useful as it ensures that in need you are able to find at once all affected resources.

Reply:
This makes sense .. I renamed the label here 623f880

Reply:
do you think it also makes sense to the label value also but to `false` if the value is false rather than not have any label at all?

Because it could be confusing for example, to know if it's an old deployment of the chart that didn't have this label yet or simply the runOnDemandOnly value is set false. WDYT?

Reply:
Hello @heshamMassoud  .. this default label (with `false` value) was originally there but then with @butenkor we decided to remove it, see: https://github.com/commercetools/k8s-charts/pull/25#discussion_r196352474

I think that this label enables some feature (in this case on-demand run) so if we don't want service to run on demand it is not needed.

Also, if we have for example 20 features and each has a default value it would get too much confusing.

Reply:
no, but here is just a key-value label, that you can use for example just to filter resources in k8s? so it's not a native field on the cronjob resource.

-But how will this label be used further on?

So, all I see here, is that if this `runOnDemandOnly` value is set to true:
1. You set the `.spec.schedule.suspend` to `true`
2. Set a label on the cronjon that `runOnDemandOnly: "true"`

But how will you be able to actually "run" the job?

> Also, if we have for example 20 features and each has a default value it would get too much confusing.

Not sure what you mean here?

Reply:
As I see it, this label will be used mostly in our k8s UI where we can trigger on-deman scripts. So the k8s UI will use this label to divide jobs to on-demand jobs and cron jobs.

In k8s we can [filter by label](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors) `runOnDemandOnly=true` but also `runOnDemandOnly != true` so we do not need a default `false` value here.

And I ment that if we have for example 20 labels and all of them have a default value, every job would have at least 20 labels so it would get confusing.

Reply:
> As I see it, this label will be used mostly in our k8s UI where we can trigger on-deman scripts. So the k8s UI will use this label to divide jobs to on-demand jobs and cron jobs.

Ok, hmm, but since there is no actual relation or logic that makes the job on demand. The only logic that setting this value actually does is set the `.spec.schedule.suspend` to `true`. So IMO, I would name the chart value and the cronjob label not `runOnDemandOnly` but more along the lines `isSuspended`. This way the user of the chart can use the the flag to make the cronjob suspended regardless if he wants to run it on demand or a totally different use case, since this is a generic template and not one that should be coupled to the k8s ui project. 

Imagine the case that I want to use the cronjob chart to have a suspended job because it's not ready yet to be running on a scheduled basis, but at the same time I don't want it to be an "on demand job", since I want to run it later on a schedule rather than on demand. In this case If I want to use the chart, I would have to set the `runOnDemandOnly` to `true` (to achieve the suspended to `true`) even though I don't want it to be an "on demand job" , do you know what I mean?

All I am saying is why don't we name the label and chart value, the same as what it does exactly, which is -> set suspended flag true or false. But not on demand or not.

> In k8s we can filter by label runOnDemandOnly=true but also runOnDemandOnly != true so we do not need a default false value here.

But this also means that applying a filter like `runOnDemandOnly = false` won't work..



> And I ment that if we have for example 20 labels and all of them have a default value, every job would have at least 20 labels so it would get confusing.

ah ok, I see your point. But it would also be a lot of labels if we 20 features and all of them are set to a non-default value :)

Reply:
In case we want to just suspend the job we can use that option `suspend` which will add `suspended=true`.

Also, if we name the label `isSuspended` instead of `runOnDemandOnly`, how would you differentiate between suspended cronjobs and on-demand jobs?

Reply:
ah, ok my bad. I didn't see the `"suspended"` value. 

So now, basically, all onDemand jobs are suspended jobs. But not all suspended jobs are on demand jobs. Ok, this makes sense. 

Disregard my concerns then :)

Reply:
Still not sure about that:
But this also means that applying a filter like runOnDemandOnly = false won't work..

😄

Reply:
IMHO, 
when I filter by `runOnDemanOnly = false` or `runOnDemanOnly != true` I expect to get the same result..

Reply:
I am not sure if we need the default value here as this condition will be used only internally... What @butenkor thinks about this?

Reply:
1. This repo is OS and the charts in it should be generic enough to work for any use case not only internal special use cases.
2. Even if it's only used internally, It's still used by other developers (not only one developer). And I think any dev would expect that filtering by the two aforementioned predicates should produce the same results.
3. I find it actually more confusing to have some labels missing at all from resources rather than have an explicitly set value (false in this case, I don't want to say "default" here, since there is no default for such a field as it is not a k8s native field but a user custom KV label).

Reply:
I would go with no default for now as custom Kubernetes UI could be used for showing cron jobs based on our templates as also own templates. If one does not set for own cron job templates explicit value for `runOnDemanOnly = false` then filtering results of Kubernetes UI might have unexpected results.

Reply:
that is true.